services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/SegmentedInMemoryKeyValueStorage.java

```diff
@@ -270,6 +270,13 @@ public void put(
       removedKeys.computeIfAbsent(segmentIdentifier, __ -> new HashSet<>()).remove(Bytes.wrap(key));
     }
 
+    @Override
+    public Optional<byte[]> get(final SegmentIdentifier segmentIdentifier, final byte[] key) {
+      return updatedValues
+          .computeIfAbsent(segmentIdentifier, __ -> new HashMap<>())
+          .getOrDefault(Bytes.wrap(key), Optional.empty());
```

Comment:
Just checking my understanding...if we are computing empty HashMap if absent, then can we ever actually return Optional.empty() here?

Reply:
Do we use this InMemory class for writing data to the database (i.e. persisting this state) or only for reading data and/or unit tests?

Reply:
> Just checking my understanding...if we are computing empty HashMap if absent, then can we ever actually return Optional.empty() here?
We can because the empty hashmap won't have the key. This just ensures that we have a map for the requested segment identifier. The put and remove methods also use this pattern.